src/main/java/frc/trigon/robot/subsystems/sideshooter/SideShooter.java

```diff
@@ -0,0 +1,72 @@
+package frc.trigon.robot.subsystems.sideshooter;
+
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.trigon.robot.utilities.Conversions;
+
+public class SideShooter extends SubsystemBase {
+    private final static SideShooter INSTANCE = new SideShooter();
+    private final TalonFX shootingMotor = SideShooterConstants.SHOOTING_MOTOR;
+    private final CANSparkMax angleMotor = SideShooterConstants.ANGLE_MOTOR;
+    private final VoltageOut shootingVoltageRequest = new VoltageOut(0, SideShooterConstants.FOC_ENABLED, false);
+    private TrapezoidProfile angleMotorProfile = null;
+    private double lastAngleMotorProfileGeneration;
+
+    public static SideShooter getInstance() {
+        return INSTANCE;
+    }
+
+    private SideShooter() {
+    }
+
+    void setTargetShootingVoltage(double targetVoltage) {
+        shootingMotor.setControl(shootingVoltageRequest.withOutput(targetVoltage));
+    }
+
+    void generateAngleMotorProfile(Rotation2d targetAngle) {
+        angleMotorProfile = new TrapezoidProfile(
+                SideShooterConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAnglePosition().getDegrees(), getAngleVelocityInDegreesPerSeconds())
+        );
+
+        lastAngleMotorProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            angleMotor.stopMotor();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleProfileTime());
+        double output = SideShooterConstants.ANGLE_PID_CONTROLLER.calculate(getAnglePosition().getDegrees(), targetState.position);
+        angleMotor.setVoltage(output);
```

Comment:
calculate feedforward out here from the ArmFeedforward class (explained in the motion profiles presentation)